fontforge/fvfonts.c

```diff
@@ -1280,35 +1283,138 @@ static RefChar *InterpRefs(RefChar *base, RefChar *other, real amount, SplineCha
 	    if ( test->checked )
 		/* Do nothing */;
 	    else if ( test->unicode_enc==base->unicode_enc &&
-		    (test->unicode_enc!=-1 || strcmp(test->sc->name,base->sc->name)==0 ) )
+		    (test->unicode_enc!=-1 || strcmp(test->sc->name,base->sc->name)==0 ) ) {
 	break;
 	}
+	}
 	if ( test!=NULL ) {
 	    test->checked = true;
-	    cur = RefCharCreate();
-	    free(cur->layers);
-	    *cur = *base;
-	    cur->orig_pos = cur->sc->orig_pos;
-	    for ( i=0; i<6; ++i )
-		cur->transform[i] = base->transform[i] + amount*(other->transform[i]-base->transform[i]);
-	    cur->layers = NULL;
-	    cur->layer_cnt = 0;
-	    cur->checked = false;
-	    if ( head==NULL )
-		head = cur;
-	    else
-		last->next = cur;
-	    last = cur;
-	} else
-	    LogError( _("In character %s, could not find reference to %s\n"),
-		    sc->name, base->sc->name );
+	    if ( !check_only ) {
+		cur = RefCharCreate();
+		free(cur->layers);
+		*cur = *base;
+		cur->orig_pos = cur->sc->orig_pos;
+		for ( i=0; i<6; ++i )
+			cur->transform[i] = base->transform[i] + amount*(other->transform[i]-base->transform[i]);
```

Comment:
This comes under a "this isn't your code and may not be your problem" disclaimer. 

It seems to me like the intent of this loop structure is to iterate through each element of "base" looking for a matching "other" temporarily called `test`. At the bottom of the loop we advance `other` if it's equal to `test` because if it happens to be the first in the list it's "checked" now and we don't have to look at it again (and we're GW so any performance gain automatically trumps code complexity).

If that's right, fine, but what I don't get is why we're then using `other` in the transform interpolation itself. Why isn't it `test` in (new) line 1298? That's what we just marked `checked`. 

Maybe it only makes sense to match up the `RefChar` entries 1:1 in order when interpolating, and therefore to grab the first off the list, but in that case why bother with `test` at all? 

So I'm 85% confident that that `other` should be `test`, with the other 15% being lack of confidence I've grasped this code at all.